### ISSUE TITLE
investigate support for on-demand verbose logging

### ISSUE THREAD

**Issue by pq** [flutter/flutter-intellij#2798]
Starting an app with `flutter run --verbose` can provide useful information.  In general though the information is too much and should be filtered out.  It would be interesting to see if we could add support that would allow for toggling of verbose output on-demand.

Challenges presented by the flutter tools implementation:

1. there is no indication, given a log entry, whether it is verbose (or not) which is exacerbated by
2. verbose mode does not seem to respect `--machine` mode, logging instead right to stdout

@DanTup : do you do any special handling of verbose output in vscode?

**Comment by DanTup** [comment]
> do you do any special handling of verbose output in vscode?

Nope! We already display stdout in the console (since as you mention, there's output that goes straight to it as raw text and not `JSON`) so when running in verbose mode you just see the extra output in the debug console.

I don't think verbose logging is the only stuff that goes raw to stdout though, so if we wanted to filter it we may need to fix that so other stuff doesn't disappear?

**Comment by pq** [comment]
Thanks for confirming @DanTup!

> I don't think verbose logging is the only stuff that goes raw to stdout though, so if we wanted to filter it we may need to fix that so other stuff doesn't disappear?

Spot on.  On the surface it seems like we could make `verbose` use a machine mode and tag it's entries as such in JSON.  I haven't explored it though...
